### PR TITLE
add all task of US joined mark

### PR DESCRIPTION
**Pull request by alex96dev**
- Define the logic to determine if an activity is joined by the user.
- Update the component to display the indicator on the activity card.
- Test the functionality across different pages to ensure consistency in displaying the indicator for joined activities.
- Integrate the logic into the frontend and backend systems to handle the display of the indicator.

### CODE REVIEW

File: components/DetailsCard/index.js [hunk 1]
```diff
@@ -104,6 +104,7 @@ export default function DetailsCard() {
   return (
     <StyledDetailsCard>
       <h1>{activities.name}</h1>
+      <StyledJoinedmark>{joined.joined && <p>XX</p>}</StyledJoinedmark>
```
Comment (MerleMak) [inline components/DetailsCard/index.js:107]: This seems more complicated than it needs to be. 
Why don't you make a useState that keeps a Boolean

`const [isJoined, setIsJoined] = useState(false)`

**Review by MerleMak — CHANGES_REQUESTED** [review]
Nice job so far! 
I think 3 useStates are a bit excessive, maybe you can thing about creating one State as an object that contains all the info of the button color and text and joined true or false. 
Either way, try to go over the "joining logic" and reconsider what you really need. 
Tip: ChatGPT is a great helper for that :)

keep it up!

**Comment by alex96dev** [comment]
Hi Merle, 
I just implemented your suggestions and I noticed that putting all 3 states in just 1 state as an object produced a significant higher amount of code due to the new syntax in order to update or calling the states.
Is this the way to go anyways?

File: components/DetailsCard/index.js [hunk 2]
```diff
@@ -8,22 +8,39 @@ import { useState, useEffect } from "react";
...
   useEffect(() => {
     const fetchData = async () => {
       try {
         const response = await fetch(`/api/activities/${id}`);
         const json = await response.json();
-        setJoined(json);
+        setJoinState((prevState) => ({
+          ...prevState,
+          isJoined: json.joined,
+        }));
```
Comment (MerleMak) [inline components/DetailsCard/index.js:25]:
```suggestion
        setJoinState((prevState) => ({
           setJoinState((prevState) => {
        const joinButtonColor = json.joined ? "red" : "green";
        const joinButtonText = json.joined ? "Disjoin" : "Join";

```

File: components/DetailsCard/index.js [hunk 3]
```diff
@@ -8,22 +8,39 @@ import { useState, useEffect } from "react";
...
-          setJoinButtonColor("red");
+          setJoinState((prevState) => ({
+            ...prevState,
+            joinButtonText: "Disjoin",
+          }));
+          setJoinState((prevState) => ({
+            ...prevState,
+            joinButtonColor: "red",
+          }));
         } else {
```
Comment (MerleMak) [inline components/DetailsCard/index.js:35]:
```suggestion
```

File: components/DetailsCard/index.js [hunk 4]
```diff
@@ -8,22 +8,39 @@ import { useState, useEffect } from "react";
...
-          setJoinButtonColor("green");
+          setJoinState((prevState) => ({
+            ...prevState,
+            joinButtonText: "Join",
+          }));
+          setJoinState((prevState) => ({
+            ...prevState,
+            joinButtonColor: "green",
+          }));
         }
```
Comment (MerleMak) [inline components/DetailsCard/index.js:44]:
```suggestion
      return {
          ...prevState,
          isJoined: json.joined,
          joinButtonColor,
          joinButtonText,
        };
      });
```

**Review by MerleMak — COMMENTED** [review]
How about this? :)

**Pull request merged**